src/System.DirectoryServices.AccountManagement/tests/ComputerPrincipalTest.cs

```diff
@@ -9,25 +9,72 @@ namespace System.DirectoryServices.AccountManagement.Tests
     public class ComputerPrincipalTest : PrincipalTest
     {
         [Fact]
+        public void Ctor_Context()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            var principal = new ComputerPrincipal(context);
+            Assert.Equal(context, principal.Context);
+            Assert.Empty(principal.ServicePrincipalNames);
+        }
+
+        [Fact]
+        public void Ctor_NullContext_ThrowsArgumentException()
+        {
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(null));
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(null, "samAcountName", "password", enabled: true));
+        }
+
+        [Fact]
+        public void Ctor_NullSamAccountName_ThrowsArgumentException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(context, null, "password", enabled: true));
+        }
+
+        [Fact]
+        public void Ctor_EmptySamAccountName_ThrowsArgumentNullException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            AssertExtensions.Throws<ArgumentNullException>("Principal.SamAccountName cannot be null or empty.", () => new ComputerPrincipal(context, string.Empty, "password", enabled: true));
+        }
+
+        [Fact]
+        public void Ctor_NullPassword_ThrowsArgumentException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            AssertExtensions.Throws<ArgumentException>(null, () => new ComputerPrincipal(context, "samAccountName", null, enabled: true));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void Ctor_MachineContext_ThrowsInvalidCastException()
+        {
+            var context = new PrincipalContext(ContextType.Machine);
+            Assert.Throws<InvalidCastException>(() => new ComputerPrincipal(context, "samAccountName", "password", enabled: true));
+        }
+
+        [Fact]
         public void ComputerPrincipalConstructorTest()
         {
+            if (DomainContext == null)
+            {
+                return;
+            }
```

Comment:
> So what I did was surround the throwing code with a catch. This allows the tests to run, but leaves DomainContext uninitialized.

How do we know if the tests are actually testing what they're supposed to test?  Are all of these tests that check this now just returning early?  If so, what's the value of the tests?